apps/aecore/src/aec_tx_processor.erl

```diff
@@ -1031,6 +1035,32 @@ assert_oracle_abi_version(ABIVersion, S) ->
         false -> runtime_error(bad_abi_version)
     end.
 
+assert_oracle_formats(_QFormat,_RFormat, ?ABI_NO_VM,_S) ->
+    %% The format strings can be anything
+    ok;
+assert_oracle_formats(QFormat, RFormat, ?ABI_SOPHIA_1, S) ->
+    case S#state.protocol >= ?MINERVA_PROTOCOL_VSN of
+        false ->
+            %% Backwards compatible: no check of this.
+            ok;
+        true ->
+            assert_typerep_format(QFormat, bad_query_format),
+            assert_typerep_format(RFormat, bad_response_format)
+    end.
+
+assert_typerep_format(Format, Error) ->
+    try aeso_heap:from_binary(typerep, Format) of
```

Comment:
I see same `{ok, TypeRep} = aeso_heap:from_binary(typerep, Format)` as in `assert_oracle_format_match`. Ok.